src/reports.coffee

```diff
@@ -168,14 +174,17 @@ htmlTemplate = (report) ->
         "
   for data in report.data
     do (data) ->
-      text += "<tr><td><i>#{data.channel.name}</i></td>"
-      text += "<td> #{ data.data[0].load } transactions </td>"
-      text += "<td> #{ data.data[0].avgResp } </td>"
-      text += "<td> #{ data.statusData[0].failed }  </td>"
-      text += "<td> #{ data.statusData[0].successful }  </td>"
-      text += "<td> #{ data.statusData[0].processing }  </td>"
-      text += "<td> #{ data.statusData[0].completed }  </td>"
-      text += "<td> #{ data.statusData[0].completedWErrors } </td></tr>"
+      if data.data[0] and data.statusData[0]
+        text += "<tr><td><i>#{data.channel.name}</i></td>"
+        text += "<td> #{ data.data[0].load } transactions </td>"
+        text += "<td> #{ data.data[0].avgResp } </td>"
+        text += "<td> #{ data.statusData[0].failed }  </td>"
+        text += "<td> #{ data.statusData[0].successful }  </td>"
+        text += "<td> #{ data.statusData[0].processing }  </td>"
+        text += "<td> #{ data.statusData[0].completed }  </td>"
+        text += "<td> #{ data.statusData[0].completedWErrors } </td></tr>"
+      else
+        text += "<tr><td>There is no data in this report</td></tr>"
```

Comment:
Instead of writing there is not data, maybe, we should rather just display values of zero. You may be able to add the null check in the string expressions above. Like `#{ if data.data[0] then data.data[0].load else 0}`